### PR TITLE
Add assert to ensure that we don't use write transactions before sync…

### PR DESCRIPTION
**Pull request by charlesmchen**
… database view registration is complete.

PTAL @michaelkirk

### CODE REVIEW

File: SignalServiceKit/src/Storage/TSStorageManager.m [hunk 1]
```diff
@@ -198,6 +291,8 @@ - (void)setupDatabaseWithSafeBlockingMigrations:(void (^_Nonnull)())safeBlocking
     [self.database registerExtension:[TSDatabaseSecondaryIndexes registerTimeStampIndex] withName:@"idx"];
     [OWSMessageReceiver syncRegisterDatabaseExtension:self.database];
 
+    setDatabaseInitialized();
```
Comment (michaelkirk) [inline SignalServiceKit/src/Storage/TSStorageManager.m]: Can you leave a comment as to why we're doing this here after the sync migrations but before the migrations and async registrations?

File: SignalServiceKit/src/Storage/TSStorageManager.m [hunk 2]
```diff
@@ -33,6 +33,91 @@
...
+
+@interface OWSDatabase : YapDatabase
+
+@end
+
+#pragma mark -
+
+@implementation OWSDatabase
+
+- (YapDatabaseConnection *)newConnection
```
Comment (michaelkirk) [inline SignalServiceKit/src/Storage/TSStorageManager.m]: suggested comment:

This clobbers the superclass implementation to include instrumentation which ensures the database is in a ready state before creating transactions.

Comment (charlesmchen) [inline SignalServiceKit/src/Storage/TSStorageManager.m]: Good point.  I've added a bunch of comments.

File: SignalServiceKit/src/Storage/TSStorageManager.m (hunk 1 above)
Comment (charlesmchen) [inline SignalServiceKit/src/Storage/TSStorageManager.m]: Well, in the absence of finding documentation that can shed light on the issue we've been seeing, this issue only seems to affect sync and not async registrations.  We've always been opening write transactions before the async registrations complete without negative consequences.

Comment (michaelkirk) [inline SignalServiceKit/src/Storage/TSStorageManager.m]: Right - I just want to capture what we believe in comments so when we revisit this later or get hit by a bus or something...

**Pull request merged**